cmake/macros/macro_deal_ii_setup_target.cmake

```diff
@@ -53,13 +53,32 @@ macro(deal_ii_setup_target _target)
   endif()
 
   #
-  # Set build type with the help of the specified keyword, or
-  # CMAKE_BUILD_TYPE:
+  # Parse the optional arguments passed to this macro. These
+  # may only contain the build type (DEBUG or RELEASE).
   #
+  set(_argn_build_type)
+  foreach (_arg ${ARGN})
```

Comment:
I don't like this `foreach` here. It would allow to use the macro with a number of repeated arguments, for example:
```
deal_ii_setup_target(foobar DEBUG DEBUG RELEASE DEBUG RELEASE)
```
This should be an error. In particular because accepting `deal_ii_setup_target(foobar DEBUG RELEASE)` is ambiguous.

Reply:
Good point. I'll check for repeated arguments. I want to allow calling `deal_ii_setup_target(foobar DEBUG MODULE)` or some such and so will have to iterate over arguments to figure out which one were specified and which ones were. But it's not difficult to check that each kind is provided only once.